src/main/java/com/surrealdb/refactor/driver/parsing/ResultParser.java

```diff
@@ -0,0 +1,63 @@
+package com.surrealdb.refactor.driver.parsing;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.surrealdb.refactor.exception.UnhandledProtocolResponse;
+import com.surrealdb.refactor.types.QueryResult;
+
+public class ResultParser {
+    // this method parses the JSON element regardless of whether it is a single element
+    // or an array of elements
+
+    public ResultParser() {}
+
+    public QueryResult[] parseResultMessage(JsonElement resultMessage) {
+        QueryResult[] processedOuterResults;
+
+        // checks to see if result is a JSON element, if not throw exception
+        if (resultMessage.isJsonObject()) {
+            JsonArray outerResultArray;
+
+            if (!resultMessage.isJsonArray()) {
```

Comment:
You have already tested that it is an object above :) So we don't need to test that it is not an array here - we already know it isn't

Reply:
I didn't think about that. very true!